internal/mqtt/mqtt.go

```diff
@@ -70,7 +106,7 @@ func (t *client) Connect() error {
 
 	client := mqtt.NewClient(opts)
 	if conn := client.Connect(); conn.Wait() && conn.Error() != nil {
-		panic(conn.Error())
+		return conn.Error()
```

Comment:
Currently we lazily connect the mqtt client on the first `.Publish` call. We should change that to connecting the mqtt broker on the service startup. Because now, if the initial connection fails, this may leave our service dangling around if `onUnexpectedConnectionLoss` is not called. We should then also remove the lazy connection in the `Publish` function